src/animalShelter/AnimalShelter.ts

```diff
@@ -28,104 +26,52 @@ export default class AnimalShelter {
-    } else {
-      console.log(
-        `Cannot change reptile limit to ${newReptileLimit} there are already ${reptilesTotal} reptiles`
-      );
-    }
-  }
-  getMammalLimit() {
-    return this._mammalLimit;
-  }
-  getReptileLimit() {
-    return this._reptileLimit;
-  }
-
   showAnimals() {
+
     console.log(this.animals);
-    console.log(`lions: ${this.lions}`);
-    console.log(`crocodiles: ${this.crocodiles}`);
-    console.log(`snakes: ${this.snakes}`);
-    console.log(`tortoise: ${this.tortoise}`);
+    console.log("mammals: " + this.getMammalsCount());
+    console.log("reptiles: " + this.getReptileCount());
+    console.log(`lions: ${this.animals.filter(x => x instanceof Lion).length}`);
+    console.log(`crocodiles: ${this.animals.filter(x => x instanceof Crocodile).length}`);
+    console.log(`squirrels: ${this.animals.filter(x => x instanceof Squirrel).length}`);
+    console.log(`snakes: ${this.animals.filter(x => x instanceof Snake).length}`);
+    console.log(`tortoise: ${this.animals.filter(x => x instanceof Tortoise).length}`);
   }
 
   addAnimal(animal: Animal): void {
-    if (animal instanceof Mammal) {
-      if (this.mammalsNumber >= this._mammalLimit) {
-        console.log(
-          "Sorry, there is no space for another mammal in the Animal Shelter!"
-        );
-        return;
-      }
 
-      if (animal instanceof Lion) {
-        if (animal.getHome() !== undefined) {
-          const family = families.find(
-            (pack) => pack.name === animal.getHome()
-          );
-          family?.removeAnimal(animal as Lion);
-          animal.setHome(undefined);
-        }
-        this.lions++;
-      } else if (animal instanceof Squirrel) {
-        this.squirrels++;
-      }
-      this.animals.push(animal);
-      animal.setHome("Animal Shelter");
-      console.log(`${animal.getName()} has been added to the Animal Shelter`);
-    } else if (animal) {
-      if (this.reptileNumbers >= this._reptileLimit) {
-        console.log(
-          "Sorry, there is no space for another reptile in the Animal Shelter!"
-        );
-        return;
-      }
-      if (animal instanceof Crocodile) {
-        if (animal.getHome() !== undefined) {
-          const pack = families.find((pack) => pack.name === animal.getHome());
-          pack?.removeAnimal(animal);
-          animal.setHome(undefined);
-        }
-        this.crocodiles++;
-      } else if (animal instanceof Tortoise) {
-        this.tortoise++;
-      } else if (animal instanceof Snake) {
-        this.snakes++;
+    if (this.getReptileCount() + 1 > this.reptileLimit) {
+      console.error(
+        "Sorry, there is no space for another reptile in the Animal Shelter!"
+      );
+      return;
+    }
+    else if (this.getMammalsCount() + 1 > this.mammalLimit) {
+      console.error(
+        "Sorry, there is no space for another mammal in the Animal Shelter!"
+      );
+      return;
+    }
+    let isRemovalValid: boolean = true;
```

Comment:
no need to set the type of the variable here, we usually only set it when the variable is initially not assigned any value or if typescript can't determine the type by itself